frontend/src/container/GridGraphLayout/Graph/WidgetGraphComponent.tsx

```diff
@@ -161,13 +164,15 @@ function WidgetGraphComponent({
 	};
 
 	useEffect(() => {
-		const visibilityStateAndLegendEntry = getGraphVisiblityStateOnDataChange(
-			data,
-			true,
-			name,
-		);
-		setGraphsVisilityStates(visibilityStateAndLegendEntry.graphVisibilityStates);
-	}, [data, name]);
+		if (canModifyChart) {
+			const visibilityStateAndLegendEntry = getGraphVisibilityStateOnDataChange(
+				data,
```

Comment:
if function is taking more than 2 params it should be object to understand the visibility of arguments

Reply:
Okay